src/uri.rs

```diff
@@ -179,6 +175,39 @@ impl Uri {
     }
 }
 
+fn parse_scheme(s: &str) -> Option<usize> {
+    s.find(':')
+}
+
+fn parse_authority(s: &str) -> Option<usize> {
+    let v: Vec<&str> = s.split("://").collect();
+    let auth = v.last().unwrap()
```

Comment:
It is only used in the block `else if s.contains("://")` (albeit, now in a method call).

Would you prefer the `fn parse_authority(s: &str) -> Option<usize>` method removed ([definition](https://github.com/M3rs/hyper/blob/refactor_uri/src/uri.rs#L182)) and inlined in place of `let auth = parse_authority(s);` ([relevant line](https://github.com/M3rs/hyper/blob/refactor_uri/src/uri.rs#L66)) so the `unwrap()` call is safer?